### PR TITLE
Added test llama-2-7b with GPTQ quant. scheme

### PR DESCRIPTION
**Pull request by lopozz**
test TheBloke/Llama-2-7B-GPTQ with pytorch backend and cuda hardware. #95

### CODE REVIEW

**Comment by IlyasMoutawwakil** [comment]
Thanks a lot for the PR ! I have few suggestions:
- let's use `TheBloke/TinyLlama-1.1B-Chat-v0.3-GPTQ` to reduce download time like in [trt-llm](https://github.com/huggingface/optimum-benchmark/blob/main/tests/configs/tensorrt_llm.yaml#L14) (this is just the smallest llama I can think of)
- let's use the compose api, ie inherit stuff from other configs instead of rewriting them like [here](https://github.com/huggingface/optimum-benchmark/blob/main/tests/configs/cpu_inference_pytorch_gpt_sweep.yaml).

**Comment by lopozz** [comment]
Hi thanks for the feedback. Should I create a default file for llama like it was done for _gpt_ or _timm_? Also I am currently working with a machine with one GPU. hence, the backend.device_ids: 0

**Comment by IlyasMoutawwakil** [comment]
no need, just reuse the ones already provided as much as possible and explicit the rest:
```yaml
defaults:
  - backend: pytorch # default backend
  # order of inheritance, last one overrides previous ones
  - _base_ # inherits from base config
  - _inference_ # inherits from inference config
  - _cuda_ # inherits from cuda config
  - _self_ # hydra 1.1 compatibility

experiment_name: cuda_inference_pytorch_gptq

backend:
  model: TheBloke/TinyLlama-1.1B-Chat-v0.3-GPTQ
  quantization_config:
    exllama_config:
      version: 2

hydra:
  sweeper:
    params:
      backend.no_weights: true,false
```

**Comment by lopozz** [comment]
Hi @IlyasMoutawwakil I just modifid the file with your feedback. Let me know if I can do something else

**Comment by IlyasMoutawwakil** [comment]
@lopozz thanks, can you also add the pip installation to the test workflow

**Comment by lopozz** [comment]
Thanks for the support @IlyasMoutawwakil, it is my first pull request, but I think I solved the issue. Also, seeing the comment https://github.com/huggingface/optimum-benchmark/pull/144#issuecomment-1981232399 I did not modify the setup.py

Main changes: pip installation of optimum and auto-gptq packages

Let me know if you have further feedback.

**Comment by IlyasMoutawwakil** [comment]
@lopozz thanks a lot for working on this.
`optimum` and `auto-gptq` are still missing from `setup.py`'s extras.
I would suggest not adding `optimum` as a standalone but rather grouping both deps in `auto-gptq`
```python
    "bitsandbytes": ["bitsandbytes"],
    "auto-gptq": ["optimum", "auto-gptq"],
```
This will probably not work with ROCm 5.6 & 5.7 and CUDA 11.8 😅 as for these systems an extra url is required to download the right wheels, as explained in: https://github.com/AutoGPTQ/AutoGPTQ?tab=readme-ov-file#installation
It can be solved using `dependency_links` as explained in https://stackoverflow.com/a/30064248, but tell me if this is too complicated for the scope of your PR.

**Comment by lopozz** [comment]
@IlyasMoutawwakil 
update of the last commit:

- added 3 new identifiers in setup.py for cuda121, cuda118 and rocm to install optimum and auto-gptq according to the corresponding index urls.

I did not used dependency_links as you suggested because deprecated starting with pip version 19.0 (released 2019-01-22) https://setuptools.pypa.io/en/latest/deprecated/dependency_links.html#specifying-dependencies-that-aren-t-in-pypi-via-dependency-links. Let me know if you agree, in case I can modify it.

File: setup.py [hunk 1]
```diff
@@ -70,6 +70,9 @@
     "peft": ["peft"],
     "autoawq": ["autoawq@git+https://github.com/casper-hansen/AutoAWQ.git"],
     "bitsandbytes": ["bitsandbytes"],
+    "auto-gptq-cu118": ["optimum", "auto-gptq@https://huggingface.github.io/autogptq-index/whl/cu118/auto-gptq/auto_gptq-0.7.1%2Bcu118-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl"],
+    "auto-gptq-cu121": ["optimum", "auto-gptq"],
+    "auto-gptq-rocm": ["optimum", "auto-gptq@https://huggingface.github.io/autogptq-index/whl/rocm573/auto-gptq/auto_gptq-0.7.1%2Brocm5.7.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl"]
 }
```
Comment (IlyasMoutawwakil) [inline setup.py:76]: this won't work for rocm 5.6, let's just remove it and remove gptq tests on rocm ci by adding `and not gptq` to pytest keywords

File: Makefile [hunk 2]
```diff
@@ -15,8 +15,8 @@ CLI_MISC_REQS := testing
 
 CLI_CUDA_ONNXRUNTIME_REQS := testing,timm,diffusers
 CLI_ROCM_ONNXRUNTIME_REQS := testing,timm,diffusers
-CLI_CUDA_PYTORCH_REQS := testing,timm,diffusers,deepspeed,peft,bitsandbytes,autoawq
-CLI_ROCM_PYTORCH_REQS := testing,timm,diffusers,deepspeed,peft,autoawq
+CLI_CUDA_PYTORCH_REQS := testing,timm,diffusers,deepspeed,peft,bitsandbytes,autoawq,auto-gptq-cu118
+CLI_ROCM_PYTORCH_REQS := testing,timm,diffusers,deepspeed,peft,autoawq,auto-gptq-rocm
```
Comment (IlyasMoutawwakil) [inline Makefile:19]: no need

File: .github/workflows/test_cli_rocm_pytorch.yaml [hunk 3]
```diff
@@ -51,4 +52,4 @@ jobs:
           --device /dev/dri/renderD129
           --entrypoint /bin/bash
           opt-bench-rocm:${{ matrix.image.rocm_version }}
-          -c "pip install requests && pip install -e .[testing,diffusers,timm,deepspeed,peft,autoawq] && pytest -k 'cli and cuda and pytorch and not bnb' -x"
+          -c "pip install requests && pip install -e .[testing,diffusers,timm,deepspeed,peft,autoawq-rocm] && pytest -k 'cli and cuda and pytorch and not bnb' -x"
```
Comment (IlyasMoutawwakil) [inline .github/workflows/test_cli_rocm_pytorch.yaml:55]: let's add `and not gptq`

**Review by IlyasMoutawwakil — CHANGES_REQUESTED** [review]

**Comment by IlyasMoutawwakil** [comment]
Thanks, I left some comments, don't forget to run styling. 
There appear to be a `SIGSEGV` when testing GPTQ, probably something wrong with the cleanup code, I'll investigate the cause of that.

**Comment by IlyasMoutawwakil** [comment]
okay found why, in the code I uses `exllama_version` in `exllama_config` but it's supposed to be just `version`.
will fix that in another PR and merge it quickly.

**Comment by IlyasMoutawwakil** [comment]
fixed in #165

**Comment by IlyasMoutawwakil** [comment]
Thanks for addition @lopozz great work on your first PR 🤗

**Review by IlyasMoutawwakil — APPROVED** [review]

**Pull request merged**